test/e2e/vsn.js

```diff
@@ -0,0 +1,242 @@
+'use strict'
+
+const assert = require('assert')
+const tapePromise = require('tape-promise').default
+const tape = require('tape')
+const isRoughlyEqual = require('is-roughly-equal')
+
+const {createWhen} = require('./lib/util')
+const createClient = require('../..')
+const vsnProfile = require('../../p/vsn')
+const products = require('../../p/vsn/products')
+const {
+	line: createValidateLine,
+	journeyLeg: createValidateJourneyLeg,
+	movement: _validateMovement
+} = require('./lib/validators')
+const createValidate = require('./lib/validate-fptf-with')
+const testJourneysStationToStation = require('./lib/journeys-station-to-station')
+const testJourneysStationToAddress = require('./lib/journeys-station-to-address')
+const testJourneysStationToPoi = require('./lib/journeys-station-to-poi')
+const testEarlierLaterJourneys = require('./lib/earlier-later-journeys')
+const journeysFailsWithNoProduct = require('./lib/journeys-fails-with-no-product')
+const testDepartures = require('./lib/departures')
+const testArrivals = require('./lib/arrivals')
+
+const when = createWhen('Europe/Berlin', 'de-DE')
+
+const cfg = {
+	when,
+	products,
+	minLatitude: 47.24,
+	maxLatitude: 52.9,
+	minLongitude: -0.63,
+	maxLongitude: 14.07
+}
+
+const _validateLine = createValidateLine(cfg)
+const validateLine = (validate, l, name) => {
+	if (!l.direction) l = Object.assign({}, l, {direction: 'foo'})
```

Comment:
This code is in place because the CFL endpoint sometimes doesn't return a `direction`. Although I told you to copy the `cfl` tests, you can remove this code, unless you noticed that the `vsn` endpoint does the same.

Same with the `validateJourneyLeg` & `validateMovement` overrides.